src/functions/CertificateRequestProcessor.ts

```diff
@@ -0,0 +1,42 @@
+import { DeleteObjectCommandOutput, PutObjectCommandOutput } from '@aws-sdk/client-s3';
+import { Service } from 'typedi';
+import { validate as uuidValidate } from 'uuid';
+import { ITestResult } from '../models';
+import { ERRORS, TEST_RESULT_STATUS } from '../models/Enums';
+import { CertificateGenerationService, IGeneratedCertificateResponse } from '../services/CertificateGenerationService';
+import { CertificateUploadService } from '../services/CertificateUploadService';
+
+export type CertGenReturn = PutObjectCommandOutput | DeleteObjectCommandOutput;
+
+@Service()
+export class CertificateRequestProcessor {
+	constructor(
+		private certificateGenerationService: CertificateGenerationService,
+		private certificateUploadService: CertificateUploadService
+	) {}
+
+	public async process(testResult: ITestResult): Promise<CertGenReturn> {
+		const isCancelled = testResult.testStatus === TEST_RESULT_STATUS.CANCELLED;
+		if (isCancelled) {
+			return this.remove(testResult);
+		}
+
+		const isValid = uuidValidate(testResult.testResultId);
+		if (isValid) {
+			return this.create(testResult);
+		}
+
+		console.error(`${ERRORS.TESTRESULT_ID}`, testResult.testResultId);
+		throw new Error(`Bad Test Record: ${testResult.testResultId}`);
+	}
+
+	private async remove(testResult: ITestResult): Promise<DeleteObjectCommandOutput> {
+		return this.certificateUploadService.removeCertificate(testResult);
+	}
+
+	private async create(testResult: ITestResult): Promise<PutObjectCommandOutput> {
```

Comment:
It may be personal preference and appreciate it is the same amount of code, but I think as a team we we are trying to move away from `.then` / `.catch` in favour of using awaits, for readability.

```
  const response: IGeneratedCertificateResponse = await this.certificateGenerationService.generateCertificate(testResult);
  return this.certificateUploadService.uploadCertificate(response);
```

Reply:
Oh yeah absolutely in favour of removing promises +1